src/imageview.cpp

```diff
@@ -571,10 +599,13 @@ void ImageView::drawArrow(QPainter &painter,
                           const QPoint &start,
                           const QPoint &end,
                           qreal tipAngle,
-                          int tipLen) const
+                          int tipLen,
+                          QList<QGraphicsItem *> *annotations) const
 {
```

Comment:
Let's hope this is the last change before merging -- but no promise ;)

The extra variable `QList<QGraphicsItem *> *annotations` seems a little weird and isn't needed here; `annotations` is already a private variable. Instead, you could just drop `const` and use `annotations` directly.

Reply:
the const has been removed

the progress of making the code better and better is a kind of pleasure~ feel free to point out anything that need to be improved~